### DISCUSSION TITLE
Filter out commits from notifications

### DISCUSSION THREAD

**Discussion by driesvints** [community/community#10525] [Pull Requests]
At the moment, whenever new commits are pushed to a PR, a new notification is sent out to people subscribed to the PR. This can lead to a lot of noise in a user's notifications. Additionally, notifications are also sent out when CI tools push new code (in our case StyleCI). [Also see this feedback.](https://github.com/github/feedback/discussions/9663) I believe, in most situations, notifying users of new commits isn't very useful. GitHub has specific tooling to draw a user's attention to review code: review requests, mentions, comments. Informing users of every new commit is, in my humble opinion, overkill.

I'm a real power user of notifications and use them every single day to keep track of my work at Laravel. I've yet to encounter a situation where I wanted to be informed of new commits being pushed to a PR. This is extremely disruptive for my daily work I must say as I tend to click these notifications and then only see it's a new commit being pushed without a review being requested. 

Another annoying thing about this is that the last notification for a new PR is a direct link to the last commit pushed rather than the PR itself. Take the following example:

<img width="475" alt="Screenshot 2022-01-25 at 09 26 59" src="https://user-images.githubusercontent.com/594614/150941036-45589b6f-1359-4cb5-ab13-e99a47788dc0.png">

This was a new PR that reached my inbox [(here's the direct link of the notification)](https://github.com/laravel/framework/pull/40592/files/173158ac8e22f3eff0605b89f488dbb87c9afa65..baa3bfbb2d483c4930a4d600a3a1a76c5737b074). When I click it I'm taken to:

<img width="1344" alt="Screenshot 2022-01-25 at 09 27 08" src="https://user-images.githubusercontent.com/594614/150941105-e3a6dc22-48b8-4ea8-9946-a11455aef806.png">

Which doesn't really say anything to me. It's just the last commit made by the author to fix a styling issue. **I believe the link of a notification for a new PR should always direct to the PR conversation.**

To solve this, there's a few actions that can be taken:

1. Never send out notifications for new commits, only for other types of interactions (new PR, comments, review requests, etc)
2. Add a new setting to include these commit notifications (and disable by default)
3. Add a new setting to include these commit notifications (and enable by default)

I'd be happy to see 1. happen but I can also really appreciate 2. so there's still an option for people who want the current behaviour. I'd appreciate 3. as well but I'd very much look against disabling commit notifications on every single repository I'm watching... Although if really need be, I could live with it.

Two and three can be implemented as a new "Commits" toggle here:

<img width="361" alt="Screenshot 2022-01-25 at 09 40 50" src="https://user-images.githubusercontent.com/594614/150942004-4a663b76-31bf-440c-9ec6-c81519421a49.png">

Like I said, **I'd very much want to see it disabled by default.**

Of course, directly linking notifications to comments on specific parts of code can still be useful. But linking to specific commits is, like I said before, never really useful imho. 

I believe that this change will greatly help me be more productive in managing my notifications and daily work. Thanks for your consideration ❤️

**Comment by xmo-odoo** [comment]
This issue is made more frustrating by the existence of such a filter *but only for email notifications*:

![filter](https://user-images.githubusercontent.com/7571158/154241077-62168eec-9e97-474b-98ec-9bf5acfc65a1.png)

Even more so due to the complete inability of the notifications system to provide anything useful in case of force pushes: in that case it just links to an empty diff view, which it almost certainly knows is an empty diff view beforehand. This makes such notifications literally nothing but spam in a project heavily relying on force-pushing PRs.

And really feels like being mocked by Github, because in the discussions tab, there is an *invisible* (unstyled) link to the actual diff for the force push itself (not that I think it should be linked to directly as it is probably expensive, and not at all informative if e.g. the force push was done in order to rebase the PR).

**Comment by xmo-odoo** [comment]
As an addendum to the previous comment, I also want to add that not only are "commit" notifications not the most useful in general, they're utterly useless when the PR workflow is based on rewrite / in-place edition, because in that case github literally shows an empty page which says:

> We went looking everywhere, but couldn’t find those commits.
>
> Sometimes commits can disappear after a force-push. Head back to the latest changes here.

(previously reported at #5373)

**Reply by septatrix** [reply]
And it does not even mark the notification as read in that instance. One has to manually click on the "Changes" tab for it to be marked as read (or do so on the notifications page)
